link/uprobe.go

```diff
@@ -0,0 +1,275 @@
+package link
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func init() {
+	symbolsCache = internal.NewSymbolsCache()
+}
+
+var (
+	uprobeEventsPath = filepath.Join(tracefsPath, "uprobe_events")
+
+	// reUprobeSymbol is used to strip invalid characters from the uprobe symbol
+	// as they are not allowed to be used as the EVENT token in tracefs.
+	reUprobeSymbol = regexp.MustCompile("[^a-zA-Z0-9]+")
```

Comment:
Nit: `rgxUprobeSymbol` to be similar to `rgxTraceEvent`. Not sure what the community convention is here.

Reply:
not sure either, but I'll rename to `rgx*` to keep it consistent